c/src/core/codec.c

```diff
@@ -529,15 +529,15 @@ int pn_data_vfill(pn_data_t *data, const char *fmt, va_list ap)
     case 'd':
       err = pn_data_put_double(data, va_arg(ap, double));
       break;
-    case 'Z':
+    case 'Z':                   /* encode binary, must not be NULL */
```

Comment:
As much as I agree the format letters need documenting I think that doing it inline is *not* the right way to do it - IMO a single block comment above would be much easier to encompass all the different codes.